persistence/persistence-mybatis/src/main/resources/org/cbioportal/persistence/mybatis/MutationMapper.xml

```diff
@@ -151,6 +152,61 @@
         </where>
     </sql>
 
+    <sql id="whereInMultipleMolecularProfilesByAnnotation">
+        <where>
+            <if test="sampleIds != null and !sampleIds.isEmpty()">
+                mutation.SAMPLE_ID IN (
```

Comment:
same comment as before. Maybe we want to filter out the unwanted samples at the service layer so that we can cache and reuse a single query per profile.